pkg/test/kube/accessor.go

```diff
@@ -629,55 +651,477 @@ func (a *Accessor) GetUnstructured(gvr schema.GroupVersionResource, namespace, n
 func (a *Accessor) DeleteUnstructured(gvr schema.GroupVersionResource, namespace, name string) error {
-	if err := a.dynClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), name, kubeApiMeta.DeleteOptions{}); err != nil {
+	if err := a.dynamicClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), name, kubeApiMeta.DeleteOptions{}); err != nil {
 		return fmt.Errorf("failed to delete resource %v of type %v: %v", name, gvr, err)
 	}
 	return nil
 }
 
 // ApplyContents applies the given config contents using kubectl.
 func (a *Accessor) ApplyContents(namespace string, contents string) ([]string, error) {
-	return a.ctl.applyContents(namespace, contents, false)
+	return a.applyContents(namespace, contents, false)
 }
 
 // ApplyContentsDryRun applies the given config contents using kubectl with DryRun mode.
 func (a *Accessor) ApplyContentsDryRun(namespace string, contents string) ([]string, error) {
-	return a.ctl.applyContents(namespace, contents, true)
+	return a.applyContents(namespace, contents, true)
 }
 
 // Apply applies the config in the given filename using kubectl.
 func (a *Accessor) Apply(namespace string, filename string) error {
-	return a.ctl.apply(namespace, filename, false)
+	return a.apply(namespace, filename, false)
 }
 
 // ApplyDryRun applies the config in the given filename using kubectl with DryRun mode.
 func (a *Accessor) ApplyDryRun(namespace string, filename string) error {
-	return a.ctl.apply(namespace, filename, true)
+	return a.apply(namespace, filename, true)
+}
+
+// applyContents applies the given config contents using kubectl.
+func (a *Accessor) applyContents(namespace string, contents string, dryRun bool) ([]string, error) {
+	files, err := a.contentsToFileList(contents, "accessor_applyc")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := a.applyInternal(namespace, files, dryRun); err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+// apply the config in the given filename using kubectl.
+func (a *Accessor) apply(namespace string, filename string, dryRun bool) error {
+	files, err := a.fileToFileList(filename)
+	if err != nil {
+		return err
+	}
+
+	return a.applyInternal(namespace, files, dryRun)
+}
+
+func (a *Accessor) applyInternal(namespace string, files []string, dryRun bool) error {
+	for _, f := range removeEmptyFiles(files) {
+		if err := a.applyFile(namespace, f, dryRun); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *Accessor) applyFile(namespace string, file string, dryRun bool) error {
+	if dryRun {
+		scopes.CI.Infof("Applying YAML file (DryRun mode): %v", file)
+	} else {
+		scopes.CI.Infof("Applying YAML file: %v", file)
+	}
+
+	dynamicClient, err := a.clientFactory.DynamicClient()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := a.clientFactory.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+
+	// Create the options.
+	streams, _, stdout, stderr := genericclioptions.NewTestIOStreams()
```

Comment:
There is a lot of magic boilerplate here. I assume this is copy+pasted from kubectl code or something? Are there any higher level libraries that allow applying arbitrary objects?

I know the operator is doing this, so its possible we can look there for inspiration and/or code reuse